docs/guides/parallel-calls.md

```diff
@@ -0,0 +1,48 @@
+## Performing Sequence and Parallel Calls in Tailcall
+
+Tailcall's API Gateway is crafted to effortlessly manage both sequential and parallel calls, enhancing data retrieval for better performance. This guide will walk you through the steps of making sequential and parallel calls using Tailcall.
+
+### Understanding Sequential and Parallel Calls
+
+**Sequential Calls:**  In a sequential call, requests are made one after the other, and each subsequent request depends on the response of the previous one. This is akin to a step-by-step execution where the next action relies on the outcome of the preceding one.
+
+**Parallel Calls:** In parallel calls, multiple requests are made simultaneously without waiting for the completion of one before initiating the next. This approach significantly reduces the overall time required to fetch data.
+
+### Making Sequential Calls
+
+Tailcall allows for implicit sequential calls when dependencies exist between entities. Let's consider an example where we fetch user details first and then retrieve their posts
+
+```graphql
+type Query {
+  user(id: ID!): User @http(path: "/users/{{args.id}}")
+  postsForUser(id: ID!): [Post] @http(path: "/users/{{args.id}}/posts")
+}
+```
+
+ when requesting `postsForUser` the user query happens first by default, showcasing implicit sequential behavior. This means, when asking for posts, Tailcall ensures the necessary user details are fetched beforehand, streamlining the process without explicit user intervention.
+
+ ### Making Parallel Calls
+
+ Tailcall automatically identifies opportunities for parallelization based on the absence of dependencies. Let's consider an example where we want to fetch details of multiple users simultaneously
+
+ ```graphql
+ type Query {
+  user(id: ID!): User @http(path: "/users/{{args.id}}")
+  allUsers: [User] @http(path: "/users")
+}
+ ```
+
+ In this case, the `user` queries within the `allUsers` query can be executed in parallel since they don't depend on each other. Tailcall intelligently handles this parallelization without requiring explicit user intervention.
```

Comment:
Can take a more relatable example, like getting users for the comments for a post